From: Jiri Slaby <jslaby@suse.cz>
Date: Wed, 10 Jun 2020 13:43:00 +0200
Subject: syscalls: fix offset type of ksys_ftruncate
Patch-mainline: Submitted on 2020/06/10
References: bsc#1172699

After the commit below, truncate on i586 uses ksys_ftruncate. But
ksys_ftruncate truncates the offset to unsigned long. So switch the type
of offset to loff_t which is what the lower do_sys_ftruncate expects.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
Fixes: 121b32a58a3a (x86/entry/32: Use IA32-specific wrappers for syscalls taking 64-bit arguments)
Cc: Brian Gerst <brgerst@gmail.com>
Cc: Thomas Gleixner <tglx@linutronix.de>
Cc: Dominik Brodowski <linux@dominikbrodowski.net>
---
 include/linux/syscalls.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/linux/syscalls.h
+++ b/include/linux/syscalls.h
@@ -1358,7 +1358,7 @@ static inline long ksys_lchown(const cha
 
 extern long do_sys_ftruncate(unsigned int fd, loff_t length, int small);
 
-static inline long ksys_ftruncate(unsigned int fd, unsigned long length)
+static inline long ksys_ftruncate(unsigned int fd, loff_t length)
 {
 	return do_sys_ftruncate(fd, length, 1);
 }
